src/crh/pedersen/mod.rs

```diff
@@ -120,6 +120,49 @@ impl<C: ProjectiveCurve, W: Window> FixedLengthCRH for CRH<C, W> {
     }
 }
 
+impl<C: ProjectiveCurve, W: Window> TwoToOneCRH for PedersenCRH<C, W> {
+    const LEFT_INPUT_SIZE_BITS: usize = W::WINDOW_SIZE * W::NUM_WINDOWS / 2;
+    const RIGHT_INPUT_SIZE_BITS: usize = Self::LEFT_INPUT_SIZE_BITS;
+    type Output = C::Affine;
+    type Parameters = Parameters<C>;
+
+    fn setup<R: Rng>(r: &mut R) -> Result<Self::Parameters, Error> {
+        <Self as CRH>::setup(r)
+    }
+
+    /// A naive implementation method: just concat the left input and right input together
+    ///
+    /// Warning: This evaluation method is not secure if left and right input have different sizes.
+    /// For example, if `[L = {0,1,1}, R = {1}]` and `[L = {0,1}, R = {1,1}]` evaluates to the same hash.
+    ///
+    /// TODO: use a more rigid evaluation method.
```

Comment:
Maybe we could just enforce that the inputs are of equal length? This is only intended for MTs, so that seems fine to do imo

Reply:
Yes! It might be the best way for now

Reply:
updated